### ISSUE TITLE
Error "No such parameter 'den_palette'" being thrown during build checks randomly

### ISSUE THREAD

**Issue by furran** [CleverRaven/Cataclysm-DDA#70197]
### Describe the bug

I've noticed that recently ( the last few days ) new pull requests have been failing checks because of the following error:

```
(all_mods)=> Backtrace emission took 2 seconds.
(all_mods)=> (continued from above) ERROR : src/mapgendata.h:244 [Result mapgendata::get_arg(const std::string &) const [Result = string_id<mapgen_palette>]] No such parameter "den_palette"
(all_mods)=> [ Previous repeated 61 times ]
(all_mods)=> 11:37:19.079 ERROR : src/mapgendata.h:244 [Result mapgendata::get_arg(const std::string &) const [Result = string_id<mapgen_palette>]] No such parameter "den_palette"
```

This is not a failed test, it's a test throwing an error.

The culprit _seems_ to be the [trail.json](https://github.com/CleverRaven/Cataclysm-DDA/blob/8350b7e06f072b6487ca26900f18b11e6e4ef472/data/json/mapgen/trail.json#L364-L384) file introduced in #67470, as it is the only place I could find related to mapgen that had "den_palette" and it matches the timeline I felt it started happening. 

I don't really know how to fix this properly.

### Attach save file

N/A

### Steps to reproduce

It is seemingly random. But chances are if you created pull requests recently, you have seen this error.

### Expected behavior

The mapgen tests do not throw errors

### Versions and configuration

N/A

### Additional context

Some PRs with this error in their checks for anyone who wants to check: #70163 #70184 #70149 #70120

**Comment by GoLoT** [comment]
I saw this too while loading a new overmap tile in a new world (created 2 days ago on latest experimental at the time). The error message was the same as in the tests and it showed up multiple times per overmap tile. I only had it happen twice so far but I didn't continue exploring.

**Comment by Maleclypse** [comment]
Fixed by #70248
